src/logic/contracts/safeContracts.ts

```diff
@@ -63,8 +64,12 @@ const getSafeContractDeployment = ({ safeVersion }: { safeVersion: string }) =>
  */
 const getGnosisSafeContractInstance = (web3: Web3, chainId: ChainId): GnosisSafe => {
   const safeSingletonDeployment = getSafeContractDeployment({ safeVersion: LATEST_SAFE_VERSION })
+  const contractAddress = safeSingletonDeployment?.networkAddresses[chainId]
+
+  if (!contractAddress) {
+    throw new Error(`GnosisSafe contract not found for chainId: ${chainId}`)
```

Comment:
Will the app crash or are these errors caught downstream?

Reply:
I think we should be safe either way. If it's caught downstream, it prevents transactions from happening and if not, it crashes the app (and is caught by the ErrorBoundary). I'll look into each in a bit more details, just in case.

Reply:
It is fair to crash the app in this case :D (better than triggering a faulty transaction ;) )